Sources/Lifecycle/Lifecycle.swift

```diff
@@ -183,6 +183,8 @@ public struct LifecycleShutdownHandler<State> {
 ///  By default, also install shutdown hooks based on `Signal` and backtraces.
 public struct ServiceLifecycle {
     private static let backtracesInstalled = AtomicBoolean(false)
+    private static let signalHandlerInstalled = AtomicBoolean(false)
+    private static var signalHandlerSources = [DispatchSourceSignal]()
```

Comment:
this is not thread safe, also why static?

Reply:
I am confused on why we need this in the firs place, aren't the shutdown hook already cancel cleanly?

Reply:
Static as I could find no way to get it mutating properly for the instance - if making installSignalHandler mutating, I could not be able to register it as a LifecycleTask.

Reply:
signalHandlerSources is only written to once by the creating thread (using the CAS (similar to the backtrace installation)) - the actual signal register/shutdown is done in the context of the LifecycleTask and would presumably be ok?

Reply:
reverted to previos simpler approach.